src/TestUtils/include/TestUtils/TestUtils.h

```diff
@@ -19,6 +19,20 @@ MATCHER(HasValue, absl::StrCat(negation ? "Has no" : "Has a", " value.")) {
   return arg.has_value();
 }
 
+MATCHER_P(HasValue, value_matcher, absl::StrCat(negation ? "Has no" : "Has a", " value.")) {
+  if (arg.has_error()) {
+    *result_listener << "Error: " << arg.error().message();
+  }
+  return arg.has_value() && ExplainMatchResult(value_matcher, arg.value(), result_listener);
+}
+
+MATCHER_P(HasValueThat, value_matcher, absl::StrCat(negation ? "Has no" : "Has a", " value.")) {
```

Comment:
Since the overload of HasValue and HasValueThat are identical could we just drop HasValueThat?

Reply:
Yes I can do that.

Reply:
Done. PTAL!